[Refactor]성향 테스트 기능 구조 리팩토링 및  홈폴더,성향테스트 폴더 공통 색상(AppColors) 적용

ghghtls opened the pull request:
### 🚀: 개요
- 성향 테스트 기능 구조 리팩토링 및  홈폴더,성향테스트 폴더 공통 색상(AppColors) 적용

### 🚀: 작업 내용
PreferenceTestService 생성: AI 분석 기능 분리

PreferenceTestRepository 생성: Firestore 저장 및 불러오기 로직 분리

ViewModel에서 service/repository 호출 방식으로 리팩토링

HashtagSelector, RecommendedPlaceListCard, RecommendedRestaurantListCard 등 UI 컴포넌트에 AppColors 색상 적용

컴포넌트 경량화 및 warning 제거

프롬프트 토큰 소비 줄이는 방향으로 변경

[ai추천받기 route에만 저장되게 변경](https://github.com/jinsung99123/travel_muse_app/pull/156/commits/13487e43202a8d6c813524c6b4a5c4d4f7a5b137)

### 📸: 실행 화면
### 📸: 실행 화면


### 🚀: 조정 파일
view_models/preference_test_view_model.dart

services/preference/preference_test_service.dart

repositories/preference/preference_test_repository.dart

views/preference/widgets/result_action_buttons.dart

views/preference/widgets/result_view.dart

views/preference/preference_loading_page.dart

views/home/recommended_place/widgets/recommended_place_list_card.dart

views/home/recommended_place/widgets/recommended_restaurant_list_card.dart

views/home/recommended_place/widgets/hashtag_selector.dart

홈폴더,preference폴더

### 개발 결과
구조 분리로 유지보수성과 확장성 향상

공통 색상 적용으로 UI 일관성 개선

컴포넌트화 및 역할 명확화 기준 적용 완료 (warning 정리, 컴포넌트화, 역할 계층 분리)

[ai추천받기 route에만 저장되게 변경](https://github.com/jinsung99123/travel_muse_app/pull/156/commits/13487e43202a8d6c813524c6b4a5c4d4f7a5b137)

### issue
Closes #155

ghghtls committed:
fix:여행 일정 프롬프트 수정

ghghtls committed:
refactor:성향테스트 계층 분리

ghghtls committed:
refactor:컬러 appColors로 변경

ghghtls committed:
fix:await 추가

ghghtls committed:
fix:unimport

jinsung99123 commented on lib/repositories/plan/plan_repository.dart:
```diff
@@ -30,7 +30,7 @@ class PlanRepository {
 - 여행 지역: $region
```
plan_repository 29줄 파란줄
preference_test_view_model 프로바이더 분리
recommended_restaurant_list_card 43줄 withopacity 파란줄
next button 임포트 파란줄
next button withopacity 파란줄
result action buttons 67줄 context 파란줄
result view detail 96줄 tolist() 파란줄
start button 임포트 파란줄
preference test page 로직 분리
이거 모두 리펙토링후 다시 푸쉬부탁드립니다!

ghghtls committed:
refactor: 성향 테스트 ViewModel/StateNotifier 구조 분리

- PreferenceTestViewModel: 질문/답변/페이지 이동 UI 로직 담당 일반 클래스 생성
- PreferenceTestStateNotifier: Firestore 저장, 분석 등 상태 관리용 Notifier 분리
- ResultActionButtons에서 저장 로직을 StateNotifier 기반으로 변경
- Provider 파일에서 ViewModel 관련 NotifierProvider 제거 및 StateNotifierProvider로 정리
- ViewModel에서 restartTest 시 StateNotifierProvider invalidate하도록 수정

ghghtls committed:
fix:파란줄,노란줄  제거

ghghtls committed:
fix:ai추천받기 route에만 저장되게 변경
